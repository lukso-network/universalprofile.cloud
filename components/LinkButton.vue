<script setup lang="ts">
type LinkButtonSize = 'small' | 'medium'

type Props = {
  link?: LinkMetadataSocialMedia
  size?: LinkButtonSize
}

withDefaults(defineProps<Props>(), {
  size: 'medium',
  link: undefined,
})
</script>

<template>
  <lukso-button
    v-if="link?.socialMediaName"
    :size="size"
    :href="link?.url"
    variant="text"
    is-link
    is-full-width
    is-icon
    custom-class="px-0"
  >
    <div class="flex w-full items-center px-1 text-left">
      <lukso-icon
        :name="`logo-${link?.socialMediaName}-mono`"
        :size="size"
      ></lukso-icon>
    </div>
  </lukso-button>
  <lukso-button
    v-else-if="link?.url"
    :size="size"
    :href="link.url"
    variant="secondary"
    is-link
    is-full-width
  >
    <div class="flex w-full items-center text-left">
      <lukso-icon name="link" :size="size" class="mr-2"></lukso-icon>
      <span
        :class="{
          'paragraph-inter-14-medium': size === 'medium',
        }"
      >
        {{ link?.title || link.url }}
      </span>
    </div>
  </lukso-button>
</template>
