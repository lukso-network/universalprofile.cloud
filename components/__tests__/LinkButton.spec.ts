import { renderSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'

import LinkButton from '@/components/LinkButton.vue'

describe('LinkButton', () => {
  it('should render empty state', async () => {
    const component = await renderSuspended(LinkButton, {})

    expect(component.html()).toMatchSnapshot()
  })

  it('should render social link', async () => {
    const component = await renderSuspended(LinkButton, {
      props: {
        link: {
          title: 'Twitter',
          url: 'https://twitter.com',
          socialMediaName: 'x',
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render custom link', async () => {
    const component = await renderSuspended(LinkButton, {
      props: {
        link: {
          title: 'Custom',
          url: 'https://example.com',
        },
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should fallback to url when no title', async () => {
    const component = await renderSuspended(LinkButton, {
      props: {
        link: {
          url: 'https://example.com',
        } as LinkMetadataSocialMedia,
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  describe('small size', () => {
    it('should render social link', async () => {
      const component = await renderSuspended(LinkButton, {
        props: {
          link: {
            title: 'Twitter',
            url: 'https://twitter.com',
            socialMediaName: 'x',
          },
          size: 'small',
        },
      })

      expect(component.html()).toMatchSnapshot()
    })

    it('should render custom link', async () => {
      const component = await renderSuspended(LinkButton, {
        props: {
          link: {
            title: 'Custom',
            url: 'https://example.com',
          },
          size: 'small',
        },
      })

      expect(component.html()).toMatchSnapshot()
    })
  })
})
