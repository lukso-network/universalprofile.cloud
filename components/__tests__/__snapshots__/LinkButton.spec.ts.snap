// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkButton > should fallback to url when no title 1`] = `
"<div id="test-wrapper">
  <lukso-button size="medium" href="https://example.com" variant="secondary" is-link="" is-full-width="">
    <div class="flex w-full items-center text-left">
      <lukso-icon name="link" size="medium" class="mr-2"></lukso-icon><span class="paragraph-inter-14-medium">https://example.com</span>
    </div>
  </lukso-button>
</div>"
`;

exports[`LinkButton > should render custom link 1`] = `
"<div id="test-wrapper">
  <lukso-button size="medium" href="https://example.com" variant="secondary" is-link="" is-full-width="">
    <div class="flex w-full items-center text-left">
      <lukso-icon name="link" size="medium" class="mr-2"></lukso-icon><span class="paragraph-inter-14-medium">Custom</span>
    </div>
  </lukso-button>
</div>"
`;

exports[`LinkButton > should render empty state 1`] = `
"<div id="test-wrapper">
  <!--v-if-->
</div>"
`;

exports[`LinkButton > should render social link 1`] = `
"<div id="test-wrapper">
  <lukso-button size="medium" href="https://twitter.com" variant="text" is-link="" is-full-width="" is-icon="" custom-class="px-0">
    <div class="flex w-full items-center px-1 text-left">
      <lukso-icon name="logo-x-mono" size="medium"></lukso-icon>
    </div>
  </lukso-button>
</div>"
`;

exports[`LinkButton > small size > should render custom link 1`] = `
"<div id="test-wrapper">
  <lukso-button size="small" href="https://example.com" variant="secondary" is-link="" is-full-width="">
    <div class="flex w-full items-center text-left">
      <lukso-icon name="link" size="small" class="mr-2"></lukso-icon><span class="">Custom</span>
    </div>
  </lukso-button>
</div>"
`;

exports[`LinkButton > small size > should render social link 1`] = `
"<div id="test-wrapper">
  <lukso-button size="small" href="https://twitter.com" variant="text" is-link="" is-full-width="" is-icon="" custom-class="px-0">
    <div class="flex w-full items-center px-1 text-left">
      <lukso-icon name="logo-x-mono" size="small"></lukso-icon>
    </div>
  </lukso-button>
</div>"
`;
